AdyenComponents/PayTo/PayToComponent.swift

```diff
@@ -313,10 +387,115 @@ extension PayToComponent: ViewControllerPresenter {
 
 // MARK: - Event Handling
 
-extension PayToComponent {
+private extension PayToComponent {
 
-    private func didSelectContinueButton() {
+    func didSelectContinueButton() {
         // TODO: Implement
     }
+    
+    func updatePayIdIdentifier(_ newValue: String) {
+        guard let newIdentifier = PayIDIdentifier(rawValue: newValue) else { return }
+        selectedIdentifier = newIdentifier
+        selectedPaymentOption = .payID(selectedIdentifier)
+    }
+
+    func didChangeSegment(_ index: Int) {
+        
+        formViewController.view.endEditing(true)
+        switch index {
+        case 0:
+            selectedPaymentOption = .payID(selectedIdentifier)
+        case 1:
+            selectedPaymentOption = .BSB
+        default:
+            AdyenAssertion.assertionFailure(message: "Segment index out of range")
+        }
+    }
+
+}
+
+// MARK: - Private
+
+private extension PayToComponent {
+    
+    func addTopItems(to formViewController: FormViewController) {
+        let topItems: [FormItem] = [
+            flowSelectionTitleItem.padding(),
+            flowSelectionItem.padding()
+        ]
+        
+        add(topItems, to: formViewController, spacing: 1)
+    }
+    
+    func addDynamicItems(to formViewController: FormViewController) {
+        add(
+            payIdDynamicItems,
+            to: formViewController,
+            isHidden: true
+        )
+        
+        formViewController.append(paymentInstructionTitleItem)
+        formViewController.append(FormSpacerItem(numberOfSpaces: 2))
+        formViewController.append(accountNumberInputItem)
+        formViewController.append(bankStateBranchInputItem)
+        bsbDynamicItems.forEach { $0.isHidden.wrappedValue = true }
+    }
+
+    func addBottomItems(to formViewController: FormViewController) {
+        let bottomItems: [FormItem] = [
+            firstNameInputItem,
+            lastNameInputItem
+        ]
+        
+        add(bottomItems, to: formViewController)
+    }
 
+    func add(
+        _ items: [FormItem],
+        to formViewController: FormViewController,
+        spacing: Int = 0,
+        isHidden: Bool = false
+    ) {
+        items.forEach {
+            if spacing > 0 {
+                formViewController.append(FormSpacerItem(numberOfSpaces: spacing))
+            }
+            formViewController.append($0)
+            $0.isHidden.wrappedValue = isHidden
+        }
+    }
+    
+    func updateInterface() {
+        switch selectedPaymentOption {
+        case let .payID(identifier):
+            resetPayIdItemsVisibility()
+            
+            switch identifier {
+            case .phone:
```

Comment:
Just an idea. `payIdDynamicItems` could be a dictionary so this can look like
payIdDynamicItems[identifier].isHidden.wrappedValue = false
This will also simplify helper methods like `resetPayIdItemsVisibility`

Reply:
Tried it but amount of code or complexity stayed the same. I'd argue it was not better even. 
`items[.phone]?.isHidden.wrappedValue = false` is worse than `phoneItem.isHidden.wrappedValue = false`